lib/logic.rb

```diff
@@ -0,0 +1,149 @@
+class Board
+  attr_accessor :board
+
+  def initialize
+    @turn = true
+    @board = [
+      [' ', ' ', ' '],
+      [' ', ' ', ' '],
+      [' ', ' ', ' ']
+    ]
+  end
+
+  def print_board
+    @board.each do |row|
+      puts row.to_s
+    end
+  end
+
+  def full?
+    @board.each do |row|
+      row.each do |value|
+        return false if value == ' '
+      end
+    end
+    true
+  end
+
+  def taken?(row, column)
+    if @board[row][column] != ' '
+      return true
+    end
+    false
+  end
+end
+
+class Player
+  attr_accessor :name, :symbol, :board
+  def initialize(name, symbol = nil)
+    @name = name
+    @symbol = symbol
+  end
+
+  def play_at(board)
+    board.print_board
+    print 'Type your turn : '
+
+    user_input = gets.chomp 
+    case user_input
+      when '1'
+        row = 0
+        column = 0
+      when '2'
+        row = 0
+        column = 1
+      when '3'
+        row = 0
+        column = 2
+      when '4'
+        row = 1
+        column = 0
+      when '5'
+        row = 1
+        column = 1
+      when '6'
+        row = 1
+        column = 2
+      when '7'
+        row = 2
+        column = 0
+      when '8'
+        row = 2
+        column = 1
+      when '9'
+        row = 2
+        column = 2
+      else
+        puts "Thats not a valid position. Play again #{@name}"
+        row,column = play_at(board)
+    end
+    if board.taken?(row, column)
+      puts "That position is already taken by a player, please play again"
+      row,column = play_at(board)
+    else
+      puts "You played in row: #{row+1} column: #{column+1}"
+      [row, column]
+    end
+    [row, column]    
+  end
+end
+
+class Game
+  attr_accessor :symbol
+  def initialize
+    @board = Board.new
+    @player_1 = Player.new('Player 1', 'X')
+    @player_2 = Player.new('Player 2', 'O')
+    @turn = 1
+  end
+  
+  def win?(item)
```

Comment:
Is there a simpler way to find out if a player has won? As it is written right now it is a little confusing and hard to find bugs later on as this method is doing too many things